independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/Beans.java

```diff
@@ -105,17 +105,21 @@ static void processAnnotation(AnnotationInstance annotation,
             if (annotation.name()
                     .equals(DotNames.ALTERNATIVE_PRIORITY)) {
                 isAlternative = true;
-                alternativePriority = annotation.value().asInt();
+                priority = annotation.value().asInt();
                 return;
             }
             if (DotNames.DEFAULT_BEAN.equals(annotation.name())) {
                 isDefaultBean = true;
                 return;
             }
-            if (annotation.name()
-                    .equals(DotNames.PRIORITY) && alternativePriority == null) {
-                alternativePriority = annotation.value()
-                        .asInt();
+            if ((!isAlternative || priority == null) && annotation.name()
```

Comment:
`(!isAlternative || priority == null)` actually simplifies to `priority == null`, no?

Reply:
Not really, `!isAlternative` covers the use case where the priority can be set by `io.quarkus.arc.Priority`.

Reply:
But `io.quarkus.arc.Priority` is handled _below_? I mean: at this point in the code, if `isAlternative == false`, then `priority == null`, always. No?

Reply:
> But io.quarkus.arc.Priority is handled below? 

`BeanInfoBuilder.processAnnotation()` is called in no particular order.

> I mean: at this point in the code, if isAlternative == false, then priority == null, always. No?

No. `@io.quarkus.arc.Priority` could be processed before.

Reply:
OK, I admit I don't fully understand the code, so I take your word for it :-)

Reply:
It's not that complex. Hm, we simply iterate over the class annotations and for each `BeanInfoBuilder.processAnnotation()` is called. The `priority` and `isAlternative` is stored in static fields (it's not very nice... I don't get why the class is called `Builder` at all... I should probably refactor this a little bit). 

Anyway, you can have the following class:
```
@io.quarkus.arc.Priority(10)
@Priority(20)
class MyBean {}
```
Here, the priority will be set to `10` but we'd like to overwrite the value to `20`. Does it make sense?